command/fs.go

```diff
@@ -212,19 +241,121 @@ nomad alloc-status %s`, allocID, allocID)
 			)
 		}
 		f.Ui.Output(formatList(out))
-	} else {
-		// We have a file, cat it.
+		return 0
+	}
+
+	// We have a file, output it.
+	if !tail {
 		r, _, err := client.AllocFS().Cat(alloc, path, nil)
 		if err != nil {
 			f.Ui.Error(fmt.Sprintf("Error reading file: %s", err))
 			return 1
 		}
 		io.Copy(os.Stdout, r)
+	} else {
+		// Parse the offset
+		var offset int64 = int64(10) * bytesToLines
+
+		if nLines, nBytes := numLines != -1, numBytes != -1; nLines && nBytes {
+			f.Ui.Error("Both -n and -c set")
+			return 1
+		} else if nLines {
+			offset = numLines * bytesToLines
+		} else if nBytes {
+			offset = numBytes
+		}
+
+		if offset > file.Size {
+			offset = file.Size
+		}
+
+		var err error
+		if follow {
+			err = f.followFile(client, alloc, path, offset)
+		} else {
+			// This offset needs to be relative from the front versus the follow
+			// is relative to the end
+			offset = file.Size - offset
+			r, _, err := client.AllocFS().ReadAt(alloc, path, offset, -1, nil)
+			if err != nil {
+				f.Ui.Error(fmt.Sprintf("Error reading file: %s", err))
+				return 1
+			}
+			io.Copy(os.Stdout, r)
+		}
+
+		if err != nil {
+			f.Ui.Error(fmt.Sprintf("Error tailing file: %v", err))
+			return 1
+		}
 	}
 
 	return 0
 }
 
+// followFile outputs the contents of the file to stdout relative to the end of
+// the file. If numLines and numBytes are both less than zero, the default
+// output is defaulted to 10 lines.
+func (f *FSCommand) followFile(client *api.Client, alloc *api.Allocation,
+	path string, offset int64) error {
+
+	cancel := make(chan struct{})
+	frames, _, err := client.AllocFS().Stream(alloc, path, api.OriginEnd, offset, cancel, nil)
+	if err != nil {
+		return err
+	}
+	signalCh := make(chan os.Signal, 3)
```

Comment:
`3` seems pretty arbitrary... is it the max number of expected inflight signals that are in queue to be delivered?  `3` -> constant?